src/common.cpp

```diff
@@ -197,14 +197,24 @@ update_test(const char *dst_path,
 
 	struct stat src_st;
 	stat(src_path, &src_st);
-
-	if (src_st.st_mtim.tv_sec > dst_st.st_mtim.tv_sec) {
+    
+#if (defined __APPLE__)
+	if (src_st.st_mtime > dst_st.st_mtime) {
 		return true;
 	}
 
-	if (src_st.st_mtim.tv_nsec > dst_st.st_mtim.tv_nsec) {
+	if (src_st.st_mtime > dst_st.st_mtime) {
 		return true;
 	}
+#else
+    if (src_st.st_mtim.tv_sec > dst_st.st_mtim.tv_sec) {
+        return true;
+    }
+    
+    if (src_st.st_mtim.tv_nsec > dst_st.st_mtiv.tv_nsec) {
```

Comment:
Another typo: `st_mtiv` vs. `st_mtim`. If you leave out whitespace change (tabs -> spaces) finding such mistakes would be easier due to expected lack of `-` (removed/modified) lines.